scripts/rpc-maas-tool.py

```diff
@@ -13,199 +13,605 @@
+            self._write(no_alarms)
+            return 1
+
+    def excluded_checks(self):
+        """List checks that are present but shouldn't be"""
+        present_but_excluded_checks = []
+        checks = self.rpcm.get_checks(self.args.entitymatch)
+        for check in checks:
+            for exclude in self.args.excludedcheck:
+                # ping is a default check and may not have an alarm
+                if exclude in check.label:
+                    present_but_excluded_checks.append(check)
+        if present_but_excluded_checks:
+            LOGGER.info("The following checks are in the excluded_checks list"
+                        " but are still present in the API:")
+            self._write(present_but_excluded_checks)
+            return 1
+
+    def compare_checks(self):
+        """Compare checks
+
+        Check that all checks found in config files are registered with The
+        maas api.
+        """
+
+        api_checks = self.rpcm.get_checks(self.args.checkmatch)
+        missing_checks = [configcheck for configcheck in
+                          self.rpcmac.checks.keys()
+                          if configcheck not in
+                          [apicheck.label for apicheck in api_checks]]
+        if missing_checks:
+            LOGGER.info("The following checks have config files but are not "
+                        "registered with the maas api: %(missing_checks)s "
+                        % {'missing_checks': missing_checks})
+            return 1
+
+    def _compare_alarm(self, config_alarm, api_alarm):
+        """Compare one config alarm with one api alarm"""
+        return (config_alarm['label'] == api_alarm.label
+                and config_alarm['criteria'] == api_alarm.criteria)
+
+    def compare_alarms(self):
+        """Compare alarms.
+
+        Check that all alarms found in MAAS agent config files are also
+        listed by the maas api.
+        """
+        api_alarms = []
+        for entity in self.rpcm.get_entities():
+            api_alarms.extend(entity.alarms)
+
+        config_alarms = {}
+        for check in self.rpcmac.checks.values():
+            config_alarms.update(check['alarms'])
+
+        missing_alarms = []
+        for config_alarm in config_alarms.values():
+            if not any([self._compare_alarm(config_alarm, api_alarm)
+                        for api_alarm in api_alarms]):
+                missing_alarms.append(config_alarm)
+
+        if missing_alarms:
+            LOGGER.info("The following alarms are present in config files but "
+                        "are not registered with the maas api: "
+                        "%(missing_alarms)s "
+                        % {'missing_alarms':
+                            [a['label'] for a in missing_alarms]})
+            return 1
+
+    def verify_created(self):
+        """Verify that all checks and alarms have been created"""
+        LOGGER.info("--- %(datestamp)s ---"
```

Comment:
If everything if as expected this line gets added to the logs in isolation, e.g.

--- 2016-01-13 15:03:16.200597 ---
--- 2016-01-13 15:03:20.225372 ---
Checks and Alarms with failures:
 - Entity:en6D9A7qgR:aio4
   ! Check:agent.network:chAKzXzVfb:network_throughput_eth2--aio4: Completed:False Status:No such interface: eth2
     ! Alarm:alPhX1WQBP:Network transmit rate on eth2:CRITICAL
     ! Alarm:alScgsNWKx:Network receive rate on eth2:CRITICAL